core/src/main/scala/chisel3/internal/MonoConnect.scala

```diff
@@ -389,9 +389,45 @@ private[chisel3] object MonoConnect {
     _source:             Element,
     context_mod:         RawModule
   ): Unit = {
-    import BindingDirection.{Input, Internal, Output} // Using extensively so import these
+    // Reify sink and source if they're views.
     val sink = reify(_sink)
     val source = reify(_source)
+
+    checkConnect(sourceInfo, sink, source, context_mod)
+    issueConnect(sink, source)
+  }
+}
+
+private[chisel3] object checkConnect {
+  private[chisel3] def apply(
+    implicit sourceInfo: SourceInfo,
+    _sink:               Element,
+    _source:             Element,
+    context_mod:         RawModule
+  ): Unit = {
+    checkConnection(sourceInfo, _sink, _source, context_mod)
+  }
```

Comment:
```suggestion
  def apply(
    sourceInfo: SourceInfo,
    sink:               Element,
    source:             Element,
    context_mod:         RawModule
  ): Unit = {
    checkConnection(sourceInfo, sink, source, context_mod)
  }
```
No need to make the apply package private since the object already is, and the use of `implicit` and underscores in the arguments is a bit of an antipattern above so let's not propagate it to the new methods 🙂